flash/graph/GraphClassification/data.py

```diff
@@ -0,0 +1,380 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib
+from typing import Any, Callable, List, Optional, Sequence, Tuple, Union
+
+import torch
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+import networkx as nx
+from torch_geometric.data import Dataset, DataLoader
+
+
+from flash.core.classification import ClassificationDataPipeline
+from flash.core.data.datamodule import DataModule
+from flash.core.data.utils import _contains_any_tensor
+
+'''
+The structure we follow is DataSet -> DataLoader -> DataModule -> DataPipeline
+'''
+
+class BasicGraphDataset(Dataset):
```

Comment:
Not sure you need this one.

Reply:
Yeah, I'm not sure either. This is the way one would build a DataSet using the prescription from Pytorch Geometric. But perhaps it is not needed